app/Middleware/AdminAuthMiddleware.php

```
@@ -42,25 +72,193 @@ public function process(Request $request, RequestHandler $handler): Response
+            if (!$mysqli instanceof \mysqli || $mysqli->connect_errno) {
+                return null;
+            }
+
+            $mysqli->set_charset($charset);
+            self::$sharedDb = $mysqli;
+            return self::$sharedDb;
+        } catch (\Throwable $e) {
+            \App\Support\SecureLogger::error('[AdminAuthMiddleware] Failed to open DB connection for admin re-validation', ['exception' => $e->getMessage()]);
+            return null;
+        }
+    }
```

PHPMD (2.15.0)

[warning] The method getDb() has a Cyclomatic Complexity of 21. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)